### PR TITLE
Add wrong Kotlin inline function number to Android Troubleshooting

### PR DESCRIPTION
**Pull request by stefanosiano**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

added a `Kotlin` section to Android Troubleshooting regarding kotlin inline function reported with wrong line numbers
Relates to https://github.com/getsentry/sentry-java/issues/2763

### CODE REVIEW

File: src/platforms/android/troubleshooting.mdx [hunk 1]
```diff
@@ -73,6 +73,13 @@ For example, the following stack traces:
   at io.sentry.HubAdapter.captureEvent (HubAdapter.java:29)
 ```
 
+## Kotlin
+
+Stack trace elements referring to Kotlin inline functions from files different than the call site are incorrect, due to how they are compiled.
+A .class file with inline functions contains code from many source files. When the JVM generates a stacktrace, it will contain a different line number, because it doesn't substitute the class and line in which the error occurred with the original file and line number.
+As a result, if an exception is thrown inside an inline function, the line number of the stack trace of the Exception could be wrong.
+This is a missing feature in the JVM itself, so "Sentry can't fix this".
```
Comment (shanamatthews) [inline src/platforms/android/troubleshooting.mdx:81]:
```suggestion

Stack trace elements referring to Kotlin inline functions from files different than the call site may refer to the wrong line numbers.

This is due to a missing feature in the JVM. `.class` files with inline functions contain code from many source files, and when the JVM generates a stacktrace, it doesn't replace the class and line where the error occurred with the original file and line number. So, when an exception is thrown in an inline function, the line number for the stack trace of the Exception may be wrong.
```

Comment (shanamatthews) [reply src/platforms/android/troubleshooting.mdx:81]: > Stack trace elements referring to Kotlin inline functions from files different than the call site may refer to the wrong line numbers.

Is this correct? I wanted the first sentence to tell users specifically what is wrong.

**Review by shanamatthews — APPROVED** [review]

**Review by markushi — APPROVED** [review]

**Review by romtsn — APPROVED** [review]
looks good, but I'd also add:
* A link to the issue on YouTrack where it's clear that it's a missing feature on the JVM/Kotlin side
* An example with an inline function and the corresponding stacktrace, because it's a bit hard to grasp what is this about without a clear example

File: src/platforms/android/troubleshooting.mdx [hunk 2]
```diff
@@ -73,6 +73,13 @@ For example, the following stack traces:
   at io.sentry.HubAdapter.captureEvent (HubAdapter.java:29)
 ```
 
+## Kotlin
```
Comment (romtsn) [inline src/platforms/android/troubleshooting.mdx:76]:
```suggestion
## Incorrect Line Numbers in Kotlin
```

File: src/platforms/android/troubleshooting.mdx (hunk 1 above)
Comment (stefanosiano) [reply src/platforms/android/troubleshooting.mdx:81]: yep, thanks!

**Pull request merged**